### ISSUE TITLE
cancel_subscriptions can't tell that I'm closing the subscription

### ISSUE THREAD

**Issue by yjbanov** [dart-lang/linter#317]
See [example](https://github.com/flutter/flutter/blob/88e065de861e1930fc306e8e39a98a4377547a6d/packages/flutter_tools/lib/src/commands/drive.dart#L87) in Flutter.

/cc @alexeieleusis

**Comment by matanlurey** [comment]
I honestly think this lint is fundamentally flawed - yes you can whitelist some patterns that will close the subscription, but you'll need almost whole-program analysis to actually tell if something is closed - plus there are some streams (i.e. streamy) that close automatically.

**Comment by alexeieleusis** [comment]
@yjbanov did you have a chance to verify this?

**Comment by yjbanov** [comment]
Nope, but I set a reminder to verify it after the holidays.

**Comment by Hixie** [comment]
Turns out we accidentally had no lint checking at all for the past few weeks, and this isn't actually fixed.

**Comment by Hixie** [comment]
Not sure what @cbracken did there but I'm guessing that wasn't intentional.

**Comment by cbracken** [comment]
Nope! Bizarre, I actually just pushed master to cbracken/flutter master branch. Confused why that would affect this bug at all.
